### ISSUE TITLE
sqlsrv_fetch_object calls custom class constructors in static context

### ISSUE THREAD

**Issue by zburke** [Microsoft/msphpsql#139]
# Description:
When a custom classname is passed to sqlsrv_fetch_object, the constructor is called in a static context, causing an error if the constructors attempts to manipulate $this. This is inconsistent with the context used by PDO and Mysqli drivers. The problem occurs with driver version SQLSRV40.EXE under IIS 8.5. 

If you stick a print_r(debug_backtrace()) in the __construct() method of the custom class, you can see its type is '::' with sqlsrv_fetch_object instead of '->' for PDO->fetchObject or mysqli_fetch_object. 

## Test script:
```
<?php
class Foo
{
    protected $foo = null;
    public function __construct() {
        $this->foo = "bar";
    }
}
$conn = sqlsrv_connect( 'HOSTNAME', ["Database"=> 'DATABASE', "UID"=> 'USERNAME', "PWD"=>'PASSWORD']);
$stmt = sqlsrv_query($conn, "SELECT * FROM SOME_TABLE");
try {
    while ($obj = sqlsrv_fetch_object($stmt, 'Foo'))
        ;
}
catch (Error $e) {
    print_r($e);
}
```

## Expected result:
There should be no output.

## Actual result:
```
Error Object ( 
    [message:protected] => Using $this when not in object context 
    [string:Error:private] => 
    [code:protected] => 0 
    [file:protected] => W:\path\to\wwwroot\test\db.php 
    [line:protected] => 6 
    [trace:Error:private] => 
    Array ( 
        [0] => Array ( 
            [function] => __construct 
            [class] => foo 
            [type] => :: 
            [args] => Array ( ) 
        ) 
        [1] => Array ( 
            [file] => W:\path\to\wwwroot\test\db.php 
            [line] => 12 
            [function] => sqlsrv_fetch_object 
            [args] => Array ( 
                [0] => Resource id #3 [1] => foo 
            ) 
        ) 
    ) 
    [previous:Error:private] => 
)
```

**Comment by meet-bhagdev** [comment]
@zburke We are actively looking into this. Thank you for the detailed repro script.

**Comment by Hadis-Fard** [comment]
@zburke This is fixed in the [Windows 4.1.3 release](https://github.com/Microsoft/msphpsql/releases/tag/v4.1.3-Windows) and [Linux 4.0.5 release](https://github.com/Microsoft/msphpsql/releases/tag/v4.0.5-Linux). I'll close the issue, please let us know if you faced other issues.
